### PR TITLE
refactor(logstream): enhance log entry `skipProcessing` flag

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This changes the name from "needsProcessing" to "skipProcessing" which is hopefully more clear.

It also refactors the flags field of `LogEntryDescriptor` to already put everything in place for when we need to add another flag.

### CODE REVIEW

File: logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptorTest.java [hunk 1]
```diff
@@ -31,9 +31,9 @@ public void shouldMarkAsProcessed() {
     final var buffer = new UnsafeBuffer(new byte[128]);
 
     // when
-    LogEntryDescriptor.markAsProcessed(buffer, 0);
```
Comment (Zelldon) [inline logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptorTest.java:34]: :wrench: : I prefer more methods which out boolean flags, like we had before. It is often unclear what the value of the boolean does, and sometimes you have only one usecase like we do here we always set it to true.

File: logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptor.java [hunk 2]
```diff
@@ -150,16 +151,32 @@ public static void setKey(final MutableDirectBuffer buffer, final int offset, fi
...
+    return (getFlags(buffer, offset) & SKIP_PROCESSING_BIT) != 0;
+  }
+
+  public static void setSkipProcessing(
+      final MutableDirectBuffer buffer, final int offset, final boolean skip) {
+    var flags = getFlags(buffer, offset);
+    if (skip) {
+      flags |= SKIP_PROCESSING_BIT;
+    } else {
+      flags &= SKIP_PROCESSING_BIT;
```
Comment (Zelldon) [inline logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptor.java:177]: :thinking: Not sure how useful this really is.

File: logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptor.java [hunk 3]
```diff
@@ -150,16 +151,32 @@ public static void setKey(final MutableDirectBuffer buffer, final int offset, fi
...
+    buffer.putByte(flagsOffset(offset), flags);
+  }
+
+  public static boolean skipProcessing(final DirectBuffer buffer, final int offset) {
+    return (getFlags(buffer, offset) & SKIP_PROCESSING_BIT) != 0;
+  }
+
+  public static void setSkipProcessing(
+      final MutableDirectBuffer buffer, final int offset, final boolean skip) {
+    var flags = getFlags(buffer, offset);
```
Comment (Zelldon) [inline logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptor.java:173]: :thinking: Wondering why you did that now, as we discussed we haven't seen a need for now. This is not hard to add later when it is really needed. Please lets not add more code, which might be never used. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

**Review by ChrisKujawa — COMMENTED** [review]
Hey @oleschoenburg thanks for chaning the name. There are some changes I'm not sure when should do now. Lets discuss this PR on our sync.

**Review by ChrisKujawa — APPROVED** [review]
Thanks :bow:

**Pull request merged**